src/lib/helpers/helpers.ts

```diff
@@ -7,11 +7,11 @@ function distinct(array: any[], comparator?: (value1: any, value2: any) => numbe
 }
 
 function distinctWithComparator(array: any[], comparator: (value1: any, value2: any) => number): any[] {
-	return array.filter((value, index, array) => index === 0 || comparator(value, array[index - 1]) !== 0);
+	return array.sort(comparator).filter((value, index, array) => index === 0 || comparator(value, array[index - 1]) !== 0);
 }
 
 function distinctWithoutComparator(array: any[]): any[] {
-	return array.filter((value, index, array) => index === 0 || value !== array[index - 1]);
+	return array.sort().filter((value, index, array) => index === 0 || value !== array[index - 1]);
```

Comment:
I know this is convenient, but does it belong in here?

Reply:
Otherwise this function relies on the array being sorted, so I hear what you're saying, but basically the way this is working a side effect of the function is the array will come out sorted

Reply:
Actually ignore this, made a change to avoid its use

Reply:
Is it possible to remove it from both (I think I still see it on the comparator version? Git ui is a bit different to stash).